Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -2919,3 +2984,1206 @@ private command addStack pStack, pObeyDontSearch, pMarked, pIgnorePasswordProtec
+               repeat
+                  put line tSubCounter of tScriptText into tNextNonEmptyLine
+                  if the number of tokens of tNextNonEmptyLine > 0 then exit repeat
+                  add 1 to tSubCounter
+               end repeat
+               
+               if token 1 of tNextNonEmptyLine = "else" then
+                  
+                  if the number of tokens of tNextNonEmptyLine > 1 then
+                     next repeat
+                  end if
+                  
+                  if token -1 of tNextNonEmptyLine is "then" then
+                     put tCounter & cr after tCollect -- not just a one-liner
+                     next repeat
+                  end if
+                  
+                  if token 2 of tNextNonEmptyLine is "if"  \
+                        and "then" is among the tokens of tNextNonEmptyLine  \
+                        and token -1 of tNextNonEmptyLine <> "then" then
+                     next repeat
+                  end if
+                  
+                  if token -1 of tNextNonEmptyLine = "else" then
+                     put tCounter & cr after tCollect -- not just a one-liner
+                     next repeat
+                  end if
+                  
+               else
+                  next repeat -- just a one-liner
+               end if
+               
+            else -- (aLine contains "then") and not (token - 1 of aLine = "then")
+               
+               put tCounter + 1 into tSubCounter
+               -- test for "then" in line after if without then
+               repeat
+                  put line tSubCounter of tScriptText into tNextNonEmptyLine
+                  if the number of tokens of tNextNonEmptyLine > 0 then exit repeat
+                  add 1 to tSubCounter
+               end repeat
+               
+               if token 1 of tNextNonEmptyLine is "then" then
+                  if token -1 of tNextNonEmptyLine is "then" then
+                     put tCounter & cr after tCollect
+                     next repeat
+                  end if
+                  if "else" is among the words of tNextNonEmptyLine then
+                     next repeat
+                  end if
+               end if
+               
+               if token - 1 of aLine is "then" then
+                  if "else" is among the tokens of tNextNonEmptyLine  \
+                        and the number of tokens of tNextNonEmptyLine > 1 \
+                        and token 1 of tNextNonEmptyLine <> "if" then
+                     next repeat
+                  end if
+               end if
+               
+               put tCounter & cr after tCollect -- regular if
+               next repeat
+               
+            end if -- (aLine contains "then") and not (token - 1 of aLine = "then")
+         end if --token 1 of aLine is "if
+         
+         if token 1 of aLine is "else" and not (token -1 of aLine is "else") then next repeat
```

Comment:
This line prevents "else if" clauses from being recognized. Commenting it out gives a triangle to "else if" lines.